src/commands/functions-config-export.ts

```diff
@@ -0,0 +1,180 @@
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+
+import * as clc from "cli-color";
+
+import * as configExport from "../functions/runtimeConfigExport";
+import * as requireConfig from "../requireConfig";
+import { Command } from "../command";
+import { FirebaseError } from "../error";
+import { testIamPermissions } from "../gcp/iam";
+import { logger } from "../logger";
+import { resolveProjectPath } from "../projectPath";
+import { promptOnce } from "../prompt";
+import { requirePermissions } from "../requirePermissions";
+import { logBullet, logWarning, logSuccess } from "../utils";
+
+const REQUIRED_PERMISSIONS = [
+  "runtimeconfig.configs.list",
+  "runtimeconfig.configs.get",
+  "runtimeconfig.variables.list",
+  "runtimeconfig.variables.get",
+];
+
+const RESERVED_PROJECT_ALIAS = ["local"];
+
+function checkReservedAliases(pInfos: configExport.ProjectConfigInfo[]): void {
+  for (const pInfo of pInfos) {
+    if (pInfo.alias && RESERVED_PROJECT_ALIAS.includes(pInfo.alias)) {
+      logWarning(
+        `Project alias (${clc.bold(pInfo.alias)}) is reserved for internal use. ` +
+          `Saving exported config in .env.${pInfo.projectId} instead.`
+      );
+      delete pInfo.alias;
+    }
+  }
+}
+
+/* For projects where we failed to fetch the runtime config, find out what permissions are missing in the project. */
+async function checkRequiredPermission(pInfos: configExport.ProjectConfigInfo[]): Promise<void> {
+  for (const pInfo of pInfos) {
+    if (pInfo.config) continue;
+
+    const result = await testIamPermissions(pInfo.projectId, REQUIRED_PERMISSIONS);
+    if (result.passed) return;
+
+    logWarning(
+      "You are missing the following permissions to read functions config on project " +
+        `${clc.bold(pInfo.projectId)}:\n\t${result.missing.join("\n\t")}`
+    );
+
+    const confirm = await promptOnce(
+      {
+        type: "confirm",
+        name: "skip",
+        default: true,
+        message: `Continue without importing configs from project ${pInfo.projectId}?`,
+      },
+      {}
+    );
+
+    if (!confirm) {
+      throw new FirebaseError("Command aborted!");
+    }
+  }
+}
+
+async function promptForPrefix(errMsg: string): Promise<string> {
+  logWarning("The following configs keys could not be exported as environment variables:\n");
+  logWarning(errMsg);
+  return await promptOnce(
+    {
+      type: "input",
+      name: "prefix",
+      default: "CONFIG_",
+      message: "Enter a PREFIX to rename invalid environment variable keys:",
+    },
+    {}
+  );
+}
+
+async function copyFilesToDir(srcFiles: string[], destDir: string): Promise<string[]> {
+  const destFiles = [];
+  for (const file of srcFiles) {
+    const targetFile = path.join(destDir, path.basename(file));
```

Comment:
As used in this code, it does what it needs to since it is stripping the path to the tmp directly used to temporarily store the .env files, e.g. `copyFilesToDir(['/tmp/envs/.env.foo', '/tmp/envs/.env.bar'], './functions') => [./functions/.env.foo, ./functions/.env.bar]` .

Reply:
That being saiid, this code is no longer necessary.